lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php

```diff
@@ -482,6 +479,33 @@ private function buildToOneAssociationMappings(ClassMetadataInfo $metadata)
     }
 
     /**
+     * Return the asociation mapping array for the foreign key.
+     *
+     * @param \Doctrine\DBAL\Schema\ForeignKeyConstraint $foreignKey
+     * @param int                                        $columnIndex
+     *
+     * @access
+     * @return array
+     */
+    private function getForeignKeyAssociationMapping(ForeignKeyConstraint $foreignKey, $columnIndex) {
+
+        $fkColumns          = $foreignKey->getColumns();
+        $fkForeignColumns   = $foreignKey->getForeignColumns();
+
+        $associationMappingParameters = [
+            'name' => $fkColumns[$columnIndex],
```

Comment:
Would it be more efficient to pass in the `$fkColumn` and `$fkForeignColumn` rather than passing in an index and re-fetching the entire array at each call of this method?